src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -346,63 +460,19 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
 
                     result = await asyncFunc5.InvokeAsync(services2, node.IRContext.ResultType, args, _cancellationToken).ConfigureAwait(false);
                 }
-                else if (func is IAsyncConnectorTexlFunction asyncConnectorTexlFunction)
-                {
-                    return await asyncConnectorTexlFunction.InvokeAsync(args, _services, _cancellationToken).ConfigureAwait(false);
-                }
-                else if (func is CustomTexlFunction customTexlFunc)
-                {
-                    // If custom function throws an exception, don't catch it - let it propagate up to the host.
-                    result = await customTexlFunc.InvokeAsync(FunctionServices, args, _cancellationToken).ConfigureAwait(false);
-                }
-                else if (func is UserDefinedFunction userDefinedFunc)
+                else
                 {
-                    UDFStackFrame frame = new UDFStackFrame(userDefinedFunc, args);
-                    UDFStackFrame framePop = null;
-
-                    try
-                    {
-                        _udfStack.Push(frame);
-
-                        (var irnode, _) = userDefinedFunc.GetIRTranslator();
-
-                        this.CheckCancel();
-
-                        result = await irnode.Accept(this, context).ConfigureAwait(false);
-                    }
-                    finally
-                    {
-                        framePop = _udfStack.Pop();
-                    }
-
-                    if (frame != framePop)
-                    {
-                        throw new Exception("Something went wrong. UDF stack values didn't match.");
-                    }
+                    result = CommonErrors.NotYetImplementedFunctionError(node.IRContext, func.Name);
                 }
 
-#pragma warning disable CS0618 // Type or member is obsolete
-
-                // This is a temporary solution to release the Join function for host that want to use it.
-                else if (func is IAsyncTexlFunctionJoin asyncJoin)
+                if (invoker is Adapter1) // $$$ we should remove this line, so we apply this check to all impls. 

```

Comment:
>if (invoker is Adapter1)  [](http://example.com/codeflow?start=16&length=25)

This is interesting - 
before, we only did this check on certain functions. 

But we ought to do it on all functions - and are in fact failing it on toher cases (hence the need to preserver the check on 468) 

A common invoke pipeline enforces this kind of consistency,.